### PR TITLE
mixin: adhere to RFC 1123 compatible component naming

### PR DESCRIPTION
**Pull request by itspngu**
Fixes issue #4880.

* [X] I added CHANGELOG entry for this change.
* [X] Change is not relevant to the end user.

## Changes

- Renamed `bucket_replicate.libsonnet` to `bucket-replicate.libsonnet` & changed references to it accordingly
- Renamed `query_frontend.libsonnet` to `query-frontend.libsonnet` & changed references to it accordingly
- Changed component name generation & sanitization in `utils.libsonnet`
- Regenerated examples & docs

## Verification

- Ran all tests outlined in CONTRIBUTING.md and the mixin README.
- Successfully deployed the dashboards using the same approach I reported as failing in #4880.

### CODE REVIEW

**Comment by itspngu** [comment]
I'm afraid I'll need help with the failing e2e tests: https://github.com/thanos-io/thanos/runs/4280932433?check_suite_focus=true#step:5:2301

When running locally, the tests also fail (same final error), but with different runtime output. `TestQueryFrontendMemcachedCache` (failing in the linked workflow run) passes.

```
$ make test-e2e | tee test-e2e.log
all modules verified
>> building Thanos binary in /home/pngu/.asdf/installs/golang/1.17.2/packages/bin
 >   thanos
>> copying Thanos from /home/pngu/.asdf/installs/golang/1.17.2/packages/bin to ./thanos_tmp_for_docker
>> building docker image 'thanos'
[...]
>> cleaning e2e test garbage.
>> running /test/e2e tests.
# NOTE(bwplotka):
# * If you see errors on CI (timeouts), but not locally, try to add -parallel 1 to limit to single CPU to reproduce small 1CPU machine.
[...]
08:27:36 Killing minio
08:27:36 Killing minio
--- PASS: TestCompactWithStoreGatewayWithPenaltyDedup (56.89s)
--- PASS: TestCompactWithStoreGateway (57.00s)
FAIL
FAIL	github.com/thanos-io/thanos/test/e2e	57.250s
?   	github.com/thanos-io/thanos/test/e2e/e2ethanos	[no test files]
FAIL
make: *** [Makefile:244: test-e2e] Error 1

$ cat test-e2e.log | grep TestQueryFrontendMemcachedCache
=== RUN   TestQueryFrontendMemcachedCache
=== PAUSE TestQueryFrontendMemcachedCache
=== CONT  TestQueryFrontendMemcachedCache
--- PASS: TestQueryFrontendMemcachedCache (20.35s)

cat test-e2e.log | grep -B5 FAIL
    query_test.go:410: query_test.go:410:

         unexpected error: create docker network 'e2e_test_query_comp_query_0': exit status 1

=== CONT  TestReceive/hashring
--- FAIL: TestQueryCompatibilityWithPreInfoAPI (0.42s)
    --- FAIL: TestQueryCompatibilityWithPreInfoAPI/{queryImage:thanos_sidecarImage:quay.io/thanos/thanos:v0.22.0} (0.42s)
=== CONT  TestQueryFrontend
    query_frontend_test.go:30: query_frontend_test.go:30:

         unexpected error: create docker network 'e2e_test_query_frontend': exit status 1

--- FAIL: TestQueryFrontend (0.42s)
--
=== CONT  TestQuery
    query_test.go:110: query_test.go:110:

         unexpected error: create docker network 'e2e_test_query': exit status 1

--- FAIL: TestQuery (0.45s)
--
08:48:00 Killing prometheus-1
08:48:00 store-gw-1: level=info name=store-gw-1 ts=2021-11-22T08:48:00.195041309Z caller=fetcher.go:481 component=block.BaseFetcher msg="successfully synchronized block metadata" duration=7.70661ms duration_ms=7 cached=30 returned=30 partial=5
08:48:00 Killing receive-3
08:48:00 Killing receive-2
08:48:00 Killing receive-1
--- FAIL: TestReceive (0.00s)
    --- FAIL: TestReceive/multitenancy (0.42s)
    --- FAIL: TestReceive/hashring (0.16s)
--
queryAndAssert: Waiting for 34 results for query count_over_time({a="1"}[13h] offset 52490898s)
08:48:18 Killing querier-1
08:48:18 Killing store-gw-1
08:48:19 Killing minio
--- PASS: TestCompactWithStoreGateway (54.02s)
FAIL
FAIL	github.com/thanos-io/thanos/test/e2e	54.282s
?   	github.com/thanos-io/thanos/test/e2e/e2ethanos	[no test files]
FAIL
```

As a quick-and-dirty sanity check I ran the same tests on the current HEAD of main, and they fail too:

```
$ make test-e2e | tee test-e2e-main.log
>> building Thanos binary in /home/pngu/.asdf/installs/golang/1.17.2/packages/bin
 >   thanos
>> copying Thanos from /home/pngu/.asdf/installs/golang/1.17.2/packages/bin to ./thanos_tmp_for_docker
>> building docker image 'thanos'
[...]
>> cleaning e2e test garbage.
>> running /test/e2e tests.
# NOTE(bwplotka):
# * If you see errors on CI (timeouts), but not locally, try to add -parallel 1 to limit to single CPU to reproduce small 1CPU machine.
[...]
queryAndAssert: Waiting for 34 results for query count_over_time({a="1"}[13h] offset 52491480s)
08:58:00 Killing querier-1
08:58:00 Killing store-gw-1
08:58:00 Killing minio
--- PASS: TestCompactWithStoreGatewayWithPenaltyDedup (61.56s)
FAIL
FAIL	github.com/thanos-io/thanos/test/e2e	61.769s
?   	github.com/thanos-io/thanos/test/e2e/e2ethanos	[no test files]
FAIL
```

Is there something I'm missing? As I'm not familiar with the codebase and don't have too much time to sink into this I'd very much appreciate somebody pointing me in the right direction. Attaching the logs of both of my local runs to this post for good measure.

[test-e2e.log](https://github.com/thanos-io/thanos/files/7579539/test-e2e.log)
[test-e2e-main.log](https://github.com/thanos-io/thanos/files/7579540/test-e2e-main.log)

**Comment by yeya24** [comment]
Let me restart the CI. That is a known flaky test.

File: mixin/alerts/alerts.libsonnet [hunk 1]
```diff
@@ -4,6 +4,6 @@
 (import 'sidecar.libsonnet') +
 (import 'store.libsonnet') +
 (import 'rule.libsonnet') +
-(import 'bucket_replicate.libsonnet') +
+(import 'bucket-replicate.libsonnet') +
 (import 'absent.libsonnet') +
 (import 'add_runbook_links.libsonnet')
```
Comment (kakkoyun) [inline mixin/alerts/alerts.libsonnet:9]:
```suggestion
(import 'add-runbook-links.libsonnet')
```

**Review by kakkoyun — CHANGES_REQUESTED** [review]
Thanks for fixing this. Amazing work!

I didn't know about this. Could you please fix all the files in the mixin? And add a section to the readme for this. It would be even more amazing if we could have a one-liner as part of `jsonnet-lint` action https://github.com/thanos-io/thanos/blob/de0e3848ff6085acf89a5f77e053c555a2cce550/Makefile#L366-L368

File: mixin/alerts/alerts.libsonnet (hunk 1 above)
Comment (itspngu) [reply mixin/alerts/alerts.libsonnet:9]: I haven't forgotten about this, I'm just stretched pretty thin right now. I tried to make the minimum amount of changes needed to get this working - there's a lot of files with snake_case names, are you suggesting they should all be renamed?

Comment (kakkoyun) [reply mixin/alerts/alerts.libsonnet:9]: I meant to focus on the files in the `mixin` directory. For sake of consistency. I don't see any other files left besides `add-runbool-links`. We can come back to linting part later.

**Comment by itspngu** [comment]
Hey @kakkoyun I've implemented (what I think) were the desired changes. Running `make format` and `make docs` as part of the PR preparation changed some files and now the commit history is a little messy. Let me know if there's anything else to do.

**Review by kakkoyun — APPROVED** [review]

**Comment by yeya24** [comment]
@itspngu Would you mind doing a rebase on main? I think there is some conflicts but GH doesn't identify them

**Review by kakkoyun — APPROVED** [review]

**Pull request merged**